### DISCUSSION TITLE
[Feature] Sync with Apple MacOS Photos

### DISCUSSION THREAD

**Discussion by lprimak** [immich-app/immich#14842] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, I see no way to continuously sync Apple MacOS Photos library with Immich.
You can bulk upload one, but there is no way to add new photos as they are added to Photos.

I would imaging that this feature would look like this:
- All new photos that are in the MacOS Photos app would be uploaded to Immich
- All deleted photos would be removed from Immich
- All updated / edited photos would be updated in Immich

Basically, it would be the same functionality that the iOS app has, but works with Photos on Mac instead.

Thank you for the great work!

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
macOS Photos should sync with iOS Photos, so photos added that way can get synced into Immich through the iOS app. However, deletions and other changes won't be reflected to Immich.

**Reply by ndesgranges** [reply]
Yes, but you'd need an iPhone then. Mac owners do not all have iPhones (it's also my case :innocent:)

**Reply by alextran1502** [reply]
You can download the Immich app from the app store if you use M1 and above

**Reply by lprimak** [reply]
Oh really? Does it sync with apple photos?  
I didn’t know such a thing existed. Does it satisfy the requirements in this post?

**Reply by BCallifornia** [reply]
> Yes, but you'd need an iPhone then. Mac owners do not all have iPhones (it's also my case 😇)

I'm a Mac owner. HavE a Mac Mini M2 and a MacBook Air M4 13". and i also have an iPhone.

**Comment by lprimak** [comment]
There are quite a few other issues with doing it that way. Photos MacOS app doesn't transfer quite a few items to the iPhone, such as photo keywords.
When syncing directly with Photos all of those missing items can be transferred.

**Comment by mmomjian** [comment]
Duplicate: #1427

**Comment by lprimak** [comment]
Update: I was not able to find anything that correctly solves this problem, so I solved it myself.
I have created [Mirror-Immich](https://flowlogix.com/mirror-immich.html) - [GitHub Link](https://github.com/flowlogix/mirror-immich)
It creates exported symlink hierarchy, including all processed metadata that can be sync'd to Immich external library.
It also creates albums and syncs favorites optionally by using Immich API.

**Reply by bo0tzz** [reply]
@lprimak mass posting your closed source tool anywhere you can fit it is not appreciated, please stop (or open source it :P).

**Reply by lprimak** [reply]
Hi, @bo0tzz 
Thank you for your valuable feedback. I have been an open source maintainer for 10+ years (Apache Shiro, Jakarta EE, Hazelcast) and have given my precious time and energy to the open source community. I know you do the same.
I have, financially and otherwise, contributed to Immich and other open source projects.
It would be great if I were considered part of this community, like other communities that accepted me. Like I accept other members in communities that I operate in.

Open sourcing is a maintainer decision. I have reasons not to open source MirrorImmich. This may change, or not.
MirrorImmich was created to help me and my family "get a handle" on all the Apple Photos we could not before.
I released it to the public to help others do the same. Open sourced or not, it may be helpful to others and the community.

FYI: I have not collected a cent for my open source work so far. This is certainly not about money.
I hope you understand.

**Reply by bo0tzz** [reply]
You're absolutely welcome in this community, and of course open sourcing is your own decision. It's just a line we've drawn (in the past, already) that we'd prefer not to have closed-source things promoted in our spaces.
